larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.h

```diff
@@ -24,25 +29,68 @@ class TransverseMatrixTool;
 /**
  *  @brief  TwoViewTransverseTracksAlgorithm class
  */
-class TwoViewTransverseTracksAlgorithm : public NViewTrackMatchingAlgorithm<TwoViewMatchingControl<float> >
+class TwoViewTransverseTracksAlgorithm : public NViewTrackMatchingAlgorithm<TwoViewMatchingControl<TwoViewTransverseOverlapResult> >
 {
 public:
-    typedef NViewTrackMatchingAlgorithm<TwoViewMatchingControl<float> > BaseAlgorithm;
+    typedef NViewTrackMatchingAlgorithm<TwoViewMatchingControl<TwoViewTransverseOverlapResult> > BaseAlgorithm;
 
     /**
      *  @brief  Default constructor
      */
     TwoViewTransverseTracksAlgorithm();
 
 private:
-    void CalculateOverlapResult(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2, const pandora::Cluster *const);
+    /**
+     *  @brief  The overridden function called by the base class which instigates calculation of the two view overlap result
+     *
+     *  @param  pCluster1 the view 0 cluster
+     *  @param  pCluster2 the view 1 cluster
+     */
+    void CalculateOverlapResult(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2, 
+        const pandora::Cluster *const);
+
+    /**
+     *  @brief  Calculates the two view overlap result
+     *
+     *  @param  pCluster1 the view 0 cluster
+     *  @param  pCluster2 the view 1 cluster
+     *  @param  overlapResult the two view overlap result
+     */
+    pandora::StatusCode CalculateOverlapResult(const pandora::Cluster *const pCluster1, const pandora::Cluster *const pCluster2, 
+        TwoViewTransverseOverlapResult &overlapResult);
+
+    /**
+     *  @brief  Calculates the fraction of the sliding windows that contains charge bins that locally match
+     *
+     *  @param  discreteProbabilityVector1 the view 0 discrete probability vector containing the charge information
+     *  @param  pCluster2 the view 1 discrete probability vector containing the charge information
+     *  @param  randomNumberGenerator a seeded random number generator
+     *
+     *  @result the matched window fraction
+     */
+    float CalculateLocalMatchingFraction(const DiscreteProbabilityVector &discreteProbabilityVector1, 
+        const DiscreteProbabilityVector &discreteProbabilityVector2, std::mt19937 &randomNumberGenerator);
+
+    /**
+     *  @brief  Runs the list of matching tools on each element in the overlap matrix 
+     */
     void ExamineOverlapContainer();
+
+    /**
+     *  @brief  Reads the settings from the XML 
+     *
+     *  @param  xmlHandle the xml handle
+     */
```

Comment:
Some doxygen info here not required (for virtual functions originally defined in base classes).

Reply:
Got em, I Think.